### ISSUE TITLE
 Inconsistency between assembly_graph.gfa and assembly.fasta

### ISSUE THREAD

**Issue by MrTomRod** [fenderglass/ABruijn#659]
Dear Mikhail

`contig_3` in `assembly_graph.gfa` is missing in `assembly.fasta`:

```shell
$ grep ">" assembly.fasta 
>contig_1
>contig_2
```

Preview of `assembly_graph.gfa`:

```
H       VN:Z:1.0
S       edge_1  TATTTTATGAAAGAGGTGATTTTTATGCCGAAAAGA... (54kb)	dp:i:1494
S       edge_2  TCACTTATTTTCCATTCTCTTTTCATAAGATCTCTG... (3.4mb)	dp:i:451
S       edge_3  ACAAATATAATTTCATTTAAAACCTTGTCACAAAAT... (25kb)	dp:i:34
L       edge_1  +       edge_1  +       0M      RC:i:1243
L       edge_2  +       edge_2  +       0M      RC:i:343
L       edge_3  +       edge_3  +       0M      RC:i:0
P       contig_2        edge_2+ *
P       contig_1        edge_1+ *
P       contig_3        edge_3+ *
```

Visualization using `gfaviz --no-gui --render --group-labels --output assembly_graph.gfa.svg assembly_graph.gfa`:

![assembly_graph gfa](https://github.com/fenderglass/Flye/assets/40867365/82e573d8-6c4a-4ac5-a63f-6def13c2fa05)

Is that a bug?

**Comment by mikolmogorov** [comment]
Hi,

Flye applies additional coverage-based filtering to the final assembly, but not to the graph output. I think contig_3 was filtered out due to coverage. I can see that this may be a bit confusing. It's a bit tricky to synchronize, since the filter is applied after polishing procedure, which operates on contigs rather than graph.

**Comment by MrTomRod** [comment]
Thanks for the clarification!

**Comment by MrTomRod** [comment]
**TLDR:** Feature requrest: During polishing, low-coverage contigs are discarded. I think it might be better if the coverage threshold were set lower for HiFi reads vs reads with high error rates.

**Context:**

I'm working with bacterial PacBio HiFi reads. Flye finds two low-coverage (16x) plasmids but discards them during polishing. The PacBio SMRT pipeline also finds them but doesn't discard them. I think the decision of the PacBio assembler is better because HiFi reads are so accurate.

![image](https://github.com/user-attachments/assets/f4da9ae0-c4bf-4616-bff1-742bc0f6b7a6)

I went through your code a little and I think I found the relevant [filter function](https://github.com/mikolmogorov/Flye/blob/b0efd481c7bbdcdb656c830da05094044ab6f90d/flye/polishing/polish.py#L229) and [configuration file](https://github.com/mikolmogorov/Flye/blob/flye/flye/config/py_cfg.py).

If I understand correctly, Flye does not distinguish between HiFi and other reads when filtering. Do you agree that the coverage threshold should be set lower for HiFi reads vs reads with high error rates? (Is it already possible to manually set the min coverage?)

**Comment by mikolmogorov** [comment]
Thanks for the info. In the "default" mode Flye is assuming uniform coverage distribution for all all sequences, but if you are interested in plasmid assembly I recommend using the `--meta` option. It should retain contigs with variable coverage. Let me know if that works for you.

Misha

**Comment by mikolmogorov** [comment]
Assuming that this has been answered - feel free to follow up if you have more questions.
